src/main/java/no/digipost/monitoring/micrometer/ApplicationInfoMetrics.java

```diff
@@ -52,55 +54,41 @@ public class ApplicationInfoMetrics implements MeterBinder {
-        manifest = new JarManifest(classFromJar);
+    public ApplicationInfoMetrics(Class<?> classInJar) {
+        this.fromRuntimeEnvironment = KeyValueResolver
+                .inOrderOfPrecedence(
+                    FROM_SYSTEM_PROPERTIES,
+                    FROM_ENVIRONMENT_VARIABLES,
+                    Optional.ofNullable(classInJar)
+                        .flatMap(JarManifest::tryResolveFromClassInJar).or(JarManifest::tryResolveAutomatically)
+                        .map(KeyValueResolver::fromManifestMainAttributes)
+                        .orElse(noValue()));
     }
 
+
     @Override
     public void bindTo(MeterRegistry registry) {
-        fromManifestOrEnv("Implementation-Title")
-                .ifPresent(artifactId -> registry.config().commonTags("application", artifactId));
-
-        List<Tag> tags = new ArrayList<>();
-
-        addTagIfValuePresent(tags,"buildTime","Git-Build-Time");
-        addTagIfValuePresent(tags,"buildVersion","Git-Build-Version");
-        addTagIfValuePresent(tags,"buildNumber","Git-Commit");
-        addTagIfValuePresent(tags,"javaBuildVersion","Build-Jdk-Spec");
-
-        tags.add(Tag.of("javaVersion", (String) System.getProperties().get("java.version")));
+        List<Tag> tags = Stream.of(
+                fromRuntimeEnvironment.tryResolveValue("Git-Build-Time").map(buildTime -> Tag.of("buildTime", buildTime)),
+                fromRuntimeEnvironment.tryResolveValue("Git-Build-Version").map(buildVersion -> Tag.of("buildVersion", buildVersion)),
+                fromRuntimeEnvironment.tryResolveValue("Git-Commit").map(buildNumber -> Tag.of("buildNumber", buildNumber)),
+                fromRuntimeEnvironment.tryResolveValue("Build-Jdk-Spec").map(javaBuildVersion -> Tag.of("javaBuildVersion", javaBuildVersion)),
+                FROM_SYSTEM_PROPERTIES.tryResolveValue("java.version").map(javaVersion -> Tag.of("javaVersion", javaVersion)))
+            .flatMap(Optional::stream)
+            .collect(toList());
 
         Gauge.builder("app.info", () -> 1.0d)
                 .description("General build and runtime information about the application. This is a static value")
                 .tags(tags)
                 .register(registry);
     }
 
-    private void addTagIfValuePresent(List<Tag> tags, String tagKey, String valueName) {
-        fromManifestOrEnv(valueName).ifPresent(value -> tags.add(Tag.of(tagKey, value)));
-    }
-
-    private Optional<String> fromManifestOrEnv(String name) {
-        String value = environmentVariableOrSystemProperty(name);
-        if (value == null) {
-            value = manifest.getMainAttributes().getValue(name);
-        }
-        return ofNullable(value);
-    }
-
-    private static String environmentVariableOrSystemProperty(String name) {
-        String value = System.getProperty(name);
-        if (value == null) {
-            value = System.getenv(name);
-        }
-        return value;
-    }
```

Comment:
The use of these methods has been refactored to the concept of a `KeyValueResolver` which are composed of possibly multiple sources in order of precedence. This is set up in the constructor to first check system props, then environment args, and lastly the MANIFEST.MF.